notebooks/12-xarray-advanced.md

```diff
@@ -0,0 +1,686 @@
+
+<details><summary>Hints</summary>
+    
+* For the yearly average, actually both `resample(time="Y")` as `groupby("time.year")` can be used. The main difference is the returned dimension after the grouping: `resample` returns a DateTimeIndex, whereas `groupby` returns a dimension called `year` and no longer contains the DateTimeIndex. To combine the data with other time series data, using the DateTimeIndex is preferred (i.e. `resample`). 
+* `fig, ax = plt.subplots()` is a useful shortcut to prepare a Matplotlib Figure. You can add a label `set_label()` and title `set_title()` to the `axes` object.
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+ghent_pressure = era5_renamed.sel(latitude=51.05, longitude=3.71, method="nearest")["pressure_pa"]
+
+fig, ax = plt.subplots(figsize=(18, 6))
+ghent_pressure.plot.line(ax=ax)
+ghent_pressure.resample(time="Y").mean().plot.line(ax=ax)
+
+ax.set_ylabel('Pressure (Pa)')
+ax.set_title('Pressure (Pa) in Ghent (-51.05, 3.71)')
+```
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+    
+Select the precipitation data for the pixel closest to the center of Ghent (lat: -51.05, lon: 3.71) and assign the outcome to a new variable `ghent_precipitation`.
+    
+For the Ghent pixel, calculate the maximal precipitation _for each month of the year_ (1 -> 12) and convert it to mm precipitation.
+    
+To make a bar-chart (not supported in xarray) of the maximal precipitation for each month of the year, convert the output to a Pandas DataFrame using the `to_dataframe()` method and create a horizontal bar chart with the month in the y-axis and the maximal precipitation in the x-axis. Feel free to improve the axis labels.
+
+<details><summary>Hints</summary>
+    
+* You need to group based on the month of the data point, hence `groupby("...")` with `time` as an existing dimension, the `time.month` shortcut to get the corresponding month will work.
+* Plotting in Pandas is very similar to xarray, use `.plot.barh()` to create a horizontal bar chart.
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+ghent_precipitation = era5_renamed.sel(latitude=51.05, longitude=3.71, method="nearest")["precipitation_m"]
+ghent_precipitation_month_max = ghent_precipitation.groupby("time.month").max()*1000
+
+fig, ax = plt.subplots()
+ghent_precipitation_month_max.to_dataframe()["precipitation_m"].plot.barh(ax=ax)
+ax.set_xlabel("Maximal monthly precipitation (mm)")
+ax.set_ylabel("Month of the year")
+```
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+    
+Calculate the pixel-based maximal temperature _for each season_. Make a plot (`imshow`) with each of the seasons in a separate subplot next to each other. 
```

Comment:
```suggestion
Calculate the pixel-based average temperature _for each season_. Make a plot (`imshow`) with each of the seasons in a separate subplot next to each other. 
```

(to use a different reduction as in the previous exercise?)

Reply:
ok, also adjusted in the solution.